manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala

```diff
@@ -137,30 +172,72 @@ object DbClient {
       }
     }
 
-    /**
-      * Record a new queue resource in the DB.
-      *
-      * Will fail if a queue already exists with the input's name.
-      */
-    override def insertQueue(queue: Queue): IO[Unit] =
-      sql"""insert into queues (name, request_topic, response_topic, request_schema)
-          values (${queue.name}, ${queue.requestTopic}, ${queue.responseTopic}, ${queue.schema})""".update.run.void
-        .transact(transactor)
+    override def createQueue(request: QueueRequest): IO[Queue] =
+      for {
+        uuid <- FUUID.randomFUUID[IO]
+        requestTopic = s"transporter.requests.$uuid"
+        responseTopic = s"transporter.responses.$uuid"
+
+        insertStatement = sql"""insert into queues (id, name, request_topic, response_topic, request_schema)
+            values ($uuid, ${request.name}, $requestTopic, $responseTopic, ${request.schema})"""
+        _ <- insertStatement.update.run.transact(transactor)
+      } yield {
+        Queue(request.name, requestTopic, responseTopic, request.schema)
+      }
 
-    /**
-      * Remove the queue resource with the given name from the DB.
-      *
-      * No-ops if no such queue exists in the DB.
-      */
     override def deleteQueue(name: String): IO[Unit] =
       sql"""delete from queues where name = $name""".update.run.void.transact(transactor)
 
-    /** Pull the queue resource with the given name from the DB, if it exists. */
-    override def lookupQueue(name: String): IO[Option[Queue]] =
-      sql"""select name, request_topic, response_topic, request_schema
+    override def lookupQueueInfo(name: String): IO[Option[QueueInfo]] =
+      sql"""select id, request_topic, response_topic, request_schema
           from queues where name = $name"""
-        .query[Queue]
+        .query[(UUID, String, String, QueueSchema)]
         .option
         .transact(transactor)
+
+    override def recordTransferRequest(
+      queueId: UUID,
+      request: TransferRequest
+    ): IO[(UUID, List[(UUID, Json)])] =
+      for {
+        requestId <- FUUID.randomFUUID[IO]
+        requestUuid = FUUID.Unsafe.toUUID(requestId)
+        transfersById <- request.transfers.traverse { transfer =>
+          FUUID.randomFUUID[IO].map(id => FUUID.Unsafe.toUUID(id) -> transfer)
+        }
+        _ <- insertRequest(queueId, requestUuid, transfersById.map(_._1))
+          .transact(transactor)
+      } yield {
+        (requestUuid, transfersById)
+      }
+
+    override def deleteTransferRequest(id: UUID): IO[Unit] =
+      sql"""delete from transfer_requests where id = $id""".update.run.void
```

Comment:
transfer has an fk to requests right?  Does this need to cascade?

Reply:
Yes, it's set up in the liquibase migrations. [This test](https://github.com/DataBiosphere/transporter/pull/5/files#diff-a7a32a41d1669f65a9ea5a73d73b74d7R119) checks that it actually cascades.